engine/runtime/src/main/java/org/enso/interpreter/node/callable/IndirectInvokeConversionNode.java

```diff
@@ -157,9 +159,13 @@ Object doWarning(
       InvokeCallableNode.ArgumentsExecutionMode argumentsExecutionMode,
       BaseNode.TailStatus isTail,
       int thatArgumentPosition,
-      @Cached IndirectInvokeConversionNode childDispatch) {
+      @Cached IndirectInvokeConversionNode childDispatch,
+      @Cached AppendWarningNode appendWarningNode,
+      @CachedLibrary(limit = "3") WarningsLibrary warnsLib,
+      @Cached HashMapInsertAllNode mapInsertAllNode,
+      @CachedLibrary(limit = "3") InteropLibrary interop) {
     arguments[thatArgumentPosition] = that.getValue();
-    ArrayRope<Warning> warnings = that.getReassignedWarningsAsRope(this, false);
+    Warning[] warnings = that.getWarningsArray(false, warnsLib, mapInsertAllNode, interop);
```

Comment:
This should probably also operate on `EnsoHashMap` and not on `Warning[]`...

Reply:
Right. Done in a88ae7489b6778b02ac5e66266620a982b53b113.